server/src/utils/errorHandler.js

```diff
@@ -0,0 +1,23 @@
+export const statusHandler = (err, req, res, next) => {
+  console.error(err.message)
+
+  switch (err.statusCode) {
+    case 400:
+      res.status(400).json({ error: err.message, success: false })
+      break
+    case 401:
+      res.status(401).json({ error: err.message, success: false })
+      break
+    case 403:
+      res.status(403).json({ error: err.message, success: false })
+      break
+    case 404:
+      res.status(404).json({ error: err.message, success: false })
+      break
+    case 500:
+      res.status(500).json({ error: err.message, success: false })
+      break
+    default:
+      res.status(500).json({ error: err.message, success: false })
```

Comment:
It looks like there's a lot of repeated code here; the only thing that's changing at the moment is the statusCode passed to `status`. It's understandable if this is going to be expanded on and customized later, but, if not, then I'd try to get rid of the unnecessary repetition!

Reply:
Actually, this avoids the repetition. The case statement lets us throw custom errors on the frontend, pass the status code down and then send it back up. All of the errors are handled in this switch statement.

Reply:
I'm confused with the code itself if it isn't going to be expanded further, but I might just be too tired to realize how this isn't repetitive. It _looks_ like you could do the following in place of the switch statement:
```javascript
res.status(err.statusCode).json({ error: err.message, success: false })
```

Reply:
Oh yeah. Okay. So there are two reasons for this. The first is just in case we want to do additional logic based on the error type, but the second is that sometimes Mongoose sends back non http standard error codes (which won't have a code) when there are validation errors and we want to send those back as 500 errors and handle them all the same.

I've thought about doing this
```js
if (!err.statusCode) err.statusCode = 500
res.status(err.statusCode).json({ error: err.message })
```

Reply:
The non http standard is good to know, and the additional logic is definitely what validates the switch statement if nothing else!